apps/src/templates/ProtectedVisualizationDiv.jsx

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+import {connect} from 'react-redux';
+import ProtectedStatefulDiv from './ProtectedStatefulDiv';
+
+export const VISUALIZATION_DIV_ID = 'visualization';
+
+export function isResponsiveFromState(state) {
+  return !state.pageConstants.isEmbedView && !state.pageConstants.hideSource;
+}
+
+/**
+ * Protected div with ID "visualization" that depends on Redux state to render
+ * with the "responsive" class or not.
+ */
+const ProtectedVisualizationDiv = React.createClass({
+  propTypes: {
+    isResponsive: React.PropTypes.bool.isRequired
+  },
+
+  render() {
+    return (
+      <ProtectedStatefulDiv
+          id={VISUALIZATION_DIV_ID}
+          className={this.props.isResponsive ? 'responsive' : ''}
```

Comment:
Could do `className={classNames({responsive: this.props.isResponsive})}`, though I'm not sure there's a ton of value when it's just one class like this.